Jenkinsfile

```diff
@@ -37,15 +39,41 @@ Jenkins job: ${env.BUILD_URL}
 """);
 }
 
-def git_merge(branch, commit, commit_msg, merge_opts = '') {
+def git_merge(branch, commit, commit_msg) {
     sh("""\
 git config user.name jenkins
 git config user.email jenkins@example.com
 git checkout ${branch}
-git merge ${merge_opts} '${commit}' -m '${commit_msg}'
+git merge '${commit}' -m '${commit_msg}'
 """)
 }
 
+def git_rebase(branch, commit) {
+    try {
+        sh("""\
+git config user.name jenkins
+git config user.email jenkins@example.com
+git checkout ${branch}
+git rebase '${commit}'
+""")
+    } catch (err) {
+        sh("""\
+CONFLICTING_COMMIT=$(cat .git/rebase-apply/original-commit)
```

Comment:
This seems hack-ey -- is there a way to look this up using the plumbing commands?